src/lib/comms/include/sol-bluetooth.h

```diff
@@ -0,0 +1,327 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * @brief Routines to handle common Bluetooth communications.
+ *
+ * Bluetooth[1] is a standardised technology for short distance
+ * communications. Its defined by an open standard, governed by the
+ * Bluetooth Special Interest Group (SIG).
+ *
+ * Bluetooth defines two kinds of functionality, Bluetooth Smart, also
+ * referred as Bluetooth Low Energy (BLE), and Bluetooth Basic Rate.
+ * Bluetooth Basic Rate most popular application is wireless audio,
+ * Bluetooth Low Energy is becoming popular for wearable devices.
+ *
+ * [1] https://www.bluetooth.com/
+ */
+
+/**
+ * @defgroup Bluetooth Bluetooth
+ * @ingroup Comms
+ *
+ * @brief API to handle Bluetooth technology communications.
+ *
+ *
+ * @{
+ */
+
+#include <sol-network.h>
+
+/**
+ * @brief Set of types of UUIDs
+ *
+ * Bluetooth services (and other entities) types are uniquely identified
+ * by UUIDs. In Bluetooth, UUIDs come in different sizes, usually the 16-bit
+ * type is reserved to be allocated by the Bluetooth SIG. The 32-bit type is
+ * also reserved, but less used. The 128-bit is free to be used by applications.
+ */
+enum sol_bt_uuid_type {
+    SOL_BT_UUID_TYPE_16 = 2,
+    SOL_BT_UUID_TYPE_32 = 4,
+    SOL_BT_UUID_TYPE_128 = 16,
+};
+
+/**
+ * @brief Representation of a Bluetooth UUID.
+ *
+ * In Bluetooth, an UUID represents the type of an entity, for example,
+ * if a service is encountered in a remote device with the 16-bit UUID '0x111F',
+ * that service is a "HandsfreeAudioGateway".
+ *
+ * @see #sol_bt_uuid_type
+ */
+struct sol_bt_uuid {
+    enum sol_bt_uuid_type type;
+    union {
+        uint16_t val16;
+        uint32_t val32;
+        uint8_t val128[16];
+        uint8_t val[0];
+    };
+};
+
+/**
+ * @brief Convert an string to an UUID.
+ *
+ * @param uuid The uuid in which to store the value.
+ * @param str The string from which to convert.
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_bt_uuid_from_str(struct sol_bt_uuid *uuid, const struct sol_str_slice *str);
```

Comment:
If i am not mistaken, our API usually receive `const struct sol_str_slice` instead of `const struct sol_str_slice *`